src/jvm/main/org/jetbrains/kotlinx/lincheck/transformation/CodeLocations.kt

```diff
@@ -39,7 +40,7 @@ internal object CodeLocations {
     fun newCodeLocation(stackTraceElement: StackTraceElement): Int {
         val id = codeLocations.size
         codeLocations.add(stackTraceElement)
-        return id
+        return id + 1
```

Comment:
For me, using a "garbage" object looks not elegant, but if you insist, let's do it.

Reply:
BTW, why should ids start from 1, not from 0?

Reply:
Because code location 0 is standing for the method exit. I've added this as a comment.

Reply:
Why can't you use another constant for that, so you don't need to play with indices here?

Reply:
The negative values are reserved for the hashCodes of the parameters, values and receivers, used by LoopDetector. It's important for us to distingush them from the code locations to be able to find the cycle without considering them. (For example in case of a spin lock with random indexes).
See `org.jetbrains.kotlinx.lincheck.strategy.managed.LoopDetector` - the algorithm is explained in details there.

Reply:
We discussed offline not to reserve `0`, thus avoiding changing indices back and forth. In addition, we decided to remove the performance optimization of storing both hash codes and indexes with such a contract, replacing them with wrappers (sealed class?) that clearly show possible logical values (of course, if this performance optimization is not necessary).